api/metadata.py

```diff
@@ -2075,12 +2081,10 @@ def metadata_attr_list(self, request):
 
             cursor.close()
             db.close()
-
+            request_finished.send(self)
             return MetadataAttrList(items=data, count=len(data))
 
         except (IndexError, TypeError):
-            if cursor: cursor.close()
-            if db: db.close()
```

Comment:
I don't see this db.close in the finally; it should be added there to make sure the connection is closed in the case of an exception, like the cursor line is. (Unless that's not needed in Python?)